### PR TITLE
Control server UDP middleware

### PR DESCRIPTION
**Pull request by Rickebo**
Implemented control server UDP listener to forward requests received over UDP to the control server's HTTP endpoints. Implements #87.

### CODE REVIEW

File: control_server/src/middleware/udp_server.py [hunk 1]
```diff
@@ -98,17 +121,25 @@ def _handle_receive(self, data: bytes, address: Any) -> bytes:
         self.receive_event(event_data)
         return event_data.response if event_data.do_respond else None
 
+    # def _init_socket_listener(self):
```
Comment (johanlovgren) [inline control_server/src/middleware/udp_server.py:124]: Is this needed for future use?

File: control_server/src/middleware/udp_server.py [hunk 2]
```diff
@@ -98,17 +121,25 @@ def _handle_receive(self, data: bytes, address: Any) -> bytes:
...
+    # self.sock.listen()
+
     def listen(self):
         """
         Starts listening for UDP messages on a new thread.
         :return:
         """
         if self.listen_thread is None:
+            # self.socket_thread = Thread(
```
Comment (johanlovgren) [inline control_server/src/middleware/udp_server.py:133]: Is this needed for future use?

File: control_server/src/middleware/udp_server.py [hunk 3]
```diff
@@ -98,17 +121,25 @@ def _handle_receive(self, data: bytes, address: Any) -> bytes:
...
+            #     target=self._init_socket_listener,
+            # )
+
             self.listen_thread = Thread(
                 target=self._do_listen,
                 args=[]
             )
 
+        # self.socket_thread.start()
```
Comment (johanlovgren) [inline control_server/src/middleware/udp_server.py:142]: Is this needed for future use?

File: control_server/src/middleware/udp_server.py [hunk 4]
```diff
@@ -136,7 +167,11 @@ def stop(self):
...
             self.is_listening = False
+            self.sock.close()
+
+            while self.sock is not None:
+                pass
 
-        if self.sock is not None:
-            self.close()
-            self.sock = None
+        # if self.sock is not None:
```
Comment (johanlovgren) [inline control_server/src/middleware/udp_server.py:175]: Is this needed for future use?

**Review by johanlovgren — APPROVED** [review]
Looks good! 

Just some minor comments about commented code. It's maybe possible to delete it?

**Review by izaalstergren — APPROVED** [review]

**Pull request merged**